### PR TITLE
Sign and return service agreement on create

### PR DESCRIPTION
**Pull request by rupurt**
[Finishes #157726283]

### CODE REVIEW

File: cmd/remote_client_test.go [hunk 1]
```diff
@@ -80,7 +80,10 @@ func TestClient_ShowJobSpec_NotFound(t *testing.T) {
 }
 
 func TestClient_CreateServiceAgreement(t *testing.T) {
-	app, cleanup := cltest.NewApplication()
+	config, _ := cltest.NewConfig()
```
Comment (se3000) [inline cmd/remote_client_test.go]: We're now doing this regularly. Maybe make a helper for it?

File: store/models/service_agreement.go [hunk 2]
```diff
@@ -26,16 +27,30 @@ func (sa ServiceAgreement) GetID() string {
 	return sa.ID
 }
 
+// Signer is used to produce a HMAC signature from an input digest
+type Signer interface {
```
Comment (se3000) [inline store/models/service_agreement.go]: 👍

File: web/service_agreements_controller_test.go [hunk 3]
```diff
@@ -12,19 +12,31 @@ import (
...
 	}{
-		{"basic", base.String(), 200},
-		{"fails validation", base.Delete("payment").String(), 400},
-		{"invalid JSON", "{", 400},
+		{
+			"basic",
+			base.String(),
+			200,
+			"0x8de92e4d6a2b527f1e3c39022ee0f1c177a6d874224fe54f5c4c0d5bcaa57d50",
+			"0xe86c1172d713c57b8df0d9f1ae91e4524cbb4e40719798ea48d3f777d2c893c000214b70376ddb661a8dc02fcadb911582f0e00e62eece65d3f9b8d0bfb6702201",
```
Comment (se3000) [inline web/service_agreements_controller_test.go]: 🎉

File: cmd/remote_client_test.go (hunk 1 above)
Comment (rupurt) [reply cmd/remote_client_test.go]: @se3000 I think we updated this right as you were reviewing it initially.

**Pull request merged**